pkg/controllers/workgenerator/controller.go

```diff
@@ -1009,42 +1149,29 @@ func (r *Reconciler) SetupWithManager(mgr controllerruntime.Manager) error {
-							return
-						}
-					} else {
-						oldResourceSnapshot := oldWork.Labels[fleetv1beta1.ParentResourceSnapshotIndexLabel]
-						newResourceSnapshot := newWork.Labels[fleetv1beta1.ParentResourceSnapshotIndexLabel]
-						if oldResourceSnapshot == "" || newResourceSnapshot == "" {
-							klog.ErrorS(controller.NewUnexpectedBehaviorError(errors.New("found an invalid work without parent-resource-snapshot-index")),
-								"Could not find the parent resource snapshot index label", "oldWork", klog.KObj(oldWork), "oldResourceSnapshotLabelValue", oldResourceSnapshot,
-								"newWork", klog.KObj(newWork), "newResourceSnapshotLabelValue", newResourceSnapshot)
-							return
-						}
-						// There is an edge case that, the work spec is the same but from different resourceSnapshots.
-						// WorkGenerator will update the work because of the label changes, but the generation is the same.
-						// When the normal update happens, the controller will set the applied condition as false and wait
-						// until the work condition has been changed.
-						// In this edge case, we need to requeue the binding to update the binding status.
-						if oldResourceSnapshot == newResourceSnapshot {
-							klog.V(2).InfoS("The work applied or available condition stayed as true, no need to reconcile", "oldWork", klog.KObj(oldWork), "newWork", klog.KObj(newWork))
-							return
-						}
+				if !equality.Semantic.DeepEqual(oldWork.Status, newWork.Status) {
+					klog.V(2).InfoS("Work status has been changed", "oldWork", klog.KObj(oldWork), "newWork", klog.KObj(newWork))
+				} else {
+					oldResourceSnapshot := oldWork.Labels[fleetv1beta1.ParentResourceSnapshotIndexLabel]
+					newResourceSnapshot := newWork.Labels[fleetv1beta1.ParentResourceSnapshotIndexLabel]
+					if oldResourceSnapshot == "" || newResourceSnapshot == "" {
+						klog.ErrorS(controller.NewUnexpectedBehaviorError(errors.New("found an invalid work without parent-resource-snapshot-index")),
+							"Could not find the parent resource snapshot index label", "oldWork", klog.KObj(oldWork), "oldResourceSnapshotLabelValue", oldResourceSnapshot,
+							"newWork", klog.KObj(newWork), "newResourceSnapshotLabelValue", newResourceSnapshot)
+						return
+					}
+					// There is an edge case that, the work spec is the same but from different resourceSnapshots.
+					// WorkGenerator will update the work because of the label changes, but the generation is the same.
+					// When the normal update happens, the controller will set the applied condition as false and wait
+					// until the work condition has been changed.
+					// In this edge case, we need to requeue the binding to update the binding status.
+					if oldResourceSnapshot == newResourceSnapshot {
+						klog.V(2).InfoS("The work applied or available condition stayed as true, no need to reconcile", "oldWork", klog.KObj(oldWork), "newWork", klog.KObj(newWork))
+						return
 					}
 				}
```

Comment:
I wonder if we still need this?

Reply:
Yes. I am only comparing the status, so it is still necessary to have these checks that we implemented recently.